src/game.cpp

```diff
@@ -143,6 +143,12 @@ void game::tick()
     shrink_losing_player(*this);
   }
 
+  /// PASSIVE STATE ///
+  increment_passive_timers();
```

Comment:
I disagree with the refactoring of `invulnerable` into `passive`. I think the former conveys more explicit information than the latter. `passive` is too general, could refer to the player being inactive for any reason, e.g. having been killed, stunned, or recovering from a defeat, and suggest the player cannot interact with anything, including foods etc. `invulnerable` simply means the player cannot be attacked, and lose a fight

Reply:
IMO we do need to have some sound definitions of these states. For the `passive` state, I was referring to a game state that is the same as in GTA5, which means a player is alive but cannot be killed by the others players, and of course, a passive player cannot hurt the other players. A passive player can collect things and do jobs.

Considering foods or power-ups, it is worthwhile to think thoroughly. Maybe we could find a more proper name for this state I was referring to? As for `invulnerable`, I still feel like it could be a state after collecting a power-up or triggering some game mechanics.

Reply:
Fair enough, I agree with the reasoning, although again I prefer `invulnerable` over `passive` to refer to what you describe: `passive` implies you cannot do anything other than move, so no item or task collection. On the other hand, `invulnerable` does suggest you could still attack other players, which is not true (or is it?). 
I'll let you make the final call :)

Reply:
> Fair enough, I agree with the reasoning, although again I prefer `invulnerable` over `passive` to refer to what you describe: `passive` implies you cannot do anything other than move, so no item or task collection. On the other hand, `invulnerable` does suggest you could still attack other players, which is not true (or is it?). I'll let you make the final call :)

I think what is important is to have two types of invulnerable (not receiving damage) states, one you can still deal damage while the other one you cannot.

To be more specific:

- Invulnerable State A: If a player receives damage, it shrinks and becomes State A, in this state it won't receive more damage and won't deal damage or won't collide, it is like `passive` we've discussed.
- Invulnerable State B: If a player collects a power-up or something, it just won't receive any damage, but it still deals damage to the others.

Is there any possibility we find two words to well describe them?

Reply:
> Invulnerable State B: If a player collects a power-up or something, it just won't receive any damage, but it still deals damage to the others

We don't have power-ups are plans to add any before `1.0`, so let's not worry about it for now ;)
We can always make changes later if the current implementation becomes a barrier to new desired features.
Let's keep it to what's referred to in the title of this PR.
So for this "invulnerable state A", I guess either `passive` or `invulnerable` are ok names; neither is entirely satisfying but they do an ok job.

Reply:
@TheoPannetier That's true, I will refactor `passive` back to `invulnerable` for now